### PR TITLE
[GLM] perf: optimize indexer small kernels

### PR DESCRIPTION
**Pull request by yweng0828**
### What does this PR do?
Using fusion, refactoring, and other methods to optimize the small kernels in the indexer module.

### Performance
- Before optimization
image

```
                                         Detailed Performance Metrics                                         
┏━━━━━━┳━━━━━━┳━━━━━┳━━━━━━┳━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━┳━━━━━━━━┓
┃      ┃      ┃     ┃      ┃     Avg ┃     P99 ┃      Avg ┃      P99 ┃      Avg ┃      P99 ┃   Gen. ┃ Success┃
┃Conc. ┃ Rate ┃ Num ┃  RPS ┃ Lat.(s) ┃ Lat.(s) ┃ TTFT(ms) ┃ TTFT(ms) ┃ TPOT(ms) ┃ TPOT(ms) ┃ toks/s ┃    Rate┃
┡━━━━━━╇━━━━━━╇━━━━━╇━━━━━━╇━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━╇━━━━━━━━┩
│    1 │  INF │  48 │ 0.41 │   2.456 │   7.230 │    765.1 │   5411.3 │      3.4 │      4.8 │ 203.53 │  100.0%│
│    2 │  INF │  97 │ 0.54 │   3.622 │  17.510 │   1102.8 │  12344.8 │      5.0 │     20.4 │ 272.03 │  100.0%│
│    4 │  INF │  86 │ 0.57 │   6.818 │  39.400 │   1983.6 │  35320.3 │      9.7 │     55.0 │ 283.57 │  100.0%│
│    8 │  INF │ 191 │ 0.61 │  12.766 │  86.780 │   2965.4 │  74069.0 │     19.6 │    135.7 │ 303.81 │  100.0%│
│   16 │  INF │ 370 │ 0.74 │  21.045 │ 178.250 │   4619.9 │ 131873.7 │     32.9 │    303.1 │ 369.17 │  100.0%│
└──────┴──────┴─────┴──────┴─────────┴─────────┴──────────┴──────────┴──────────┴──────────┴────────┴────────┘


                                                Request Metrics                                                
┏━━━━━━┳━━━━━┳━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━━┳━━━━━━━━━━━┳━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━━━━┓
┃      ┃     ┃  Avg In ┃  P99 In ┃ Avg Out ┃ P99 Out ┃       Avg ┃    Approx ┃ Decode ┃  Decoded ┃       Spec.┃
┃Conc. ┃ Num ┃    Toks ┃    Toks ┃    Toks ┃    Toks ┃ Turns/Req ┃ Cache Hit ┃ toks/s ┃ Tok/Iter ┃ Accept Rate┃
┡━━━━━━╇━━━━━╇━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━━╇━━━━━━━━━━━╇━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━━━━┩
│    1 │  48 │ 56045.2 │ 65202.0 │   500.0 │   500.0 │      6.56 │     90.8% │ 294.99 │     3.24 │       69.1%│
│    2 │  97 │ 55744.3 │ 62610.0 │   500.0 │   500.0 │      6.63 │     91.9% │ 198.02 │     3.30 │       69.7%│
│    4 │  86 │ 54702.4 │ 62065.0 │   500.0 │   500.0 │      5.93 │     90.1% │ 103.20 │     3.29 │       69.6%│
│    8 │ 191 │ 55560.5 │ 65147.0 │   500.0 │   500.0 │      6.56 │     90.8% │  50.92 │     3.24 │       69.1%│
│   16 │ 370 │ 55488.7 │ 64581.0 │   500.0 │   500.0 │      6.38 │     91.4% │  30.38 │     3.27 │       69.4%│
└──────┴─────┴─────────┴─────────┴─────────┴─────────┴───────────┴───────────┴────────┴──────────┴────────────┘
```

- After optimization
image
Before `multi_cta_topk`, there are two `vectorized_elementwise_kernels`, corresponding to `logits.nan_to_num_` and `row_states.zero_()` respectively, which are unavoidable.

```
                                         Detailed Performance Metrics                                         
┏━━━━━━┳━━━━━━┳━━━━━┳━━━━━━┳━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━┳━━━━━━━━┓
┃      ┃      ┃     ┃      ┃     Avg ┃     P99 ┃      Avg ┃      P99 ┃      Avg ┃      P99 ┃   Gen. ┃ Success┃
┃Conc. ┃ Rate ┃ Num ┃  RPS ┃ Lat.(s) ┃ Lat.(s) ┃ TTFT(ms) ┃ TTFT(ms) ┃ TPOT(ms) ┃ TPOT(ms) ┃ toks/s ┃    Rate┃
┡━━━━━━╇━━━━━━╇━━━━━╇━━━━━━╇━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━╇━━━━━━━━┩
│    1 │  INF │  48 │ 0.42 │   2.366 │   7.050 │    762.4 │   5406.0 │      3.2 │      4.1 │ 211.24 │  100.0%│
│    2 │  INF │  97 │ 0.56 │   3.548 │  16.890 │   1118.1 │  14755.5 │      4.9 │     20.6 │ 280.06 │  100.0%│
│    4 │  INF │  86 │ 0.57 │   6.845 │  38.500 │   1966.3 │  35176.1 │      9.8 │     55.3 │ 286.64 │  100.0%│
│    8 │  INF │ 191 │ 0.62 │  12.450 │  84.610 │   3100.0 │  71774.6 │     18.7 │    128.6 │ 311.23 │  100.0%│
│   16 │  INF │ 370 │ 0.76 │  20.485 │ 177.190 │   4509.9 │ 133211.1 │     32.0 │    305.5 │ 379.91 │  100.0%│
└──────┴──────┴─────┴──────┴─────────┴─────────┴──────────┴──────────┴──────────┴──────────┴────────┴────────┘


                                                Request Metrics                                                
┏━━━━━━┳━━━━━┳━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━━┳━━━━━━━━━━━┳━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━━━━┓
┃      ┃     ┃  Avg In ┃  P99 In ┃ Avg Out ┃ P99 Out ┃       Avg ┃    Approx ┃ Decode ┃  Decoded ┃       Spec.┃
┃Conc. ┃ Num ┃    Toks ┃    Toks ┃    Toks ┃    Toks ┃ Turns/Req ┃ Cache Hit ┃ toks/s ┃ Tok/Iter ┃ Accept Rate┃
┡━━━━━━╇━━━━━╇━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━━╇━━━━━━━━━━━╇━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━━━━┩
│    1 │  48 │ 56754.4 │ 65698.0 │   500.0 │   500.0 │      6.56 │     90.7% │ 311.53 │     3.21 │       68.8%│
│    2 │  97 │ 56194.6 │ 66242.0 │   500.0 │   500.0 │      6.63 │     91.8% │ 205.34 │     3.26 │       69.3%│
│    4 │  86 │ 54915.8 │ 62035.0 │   500.0 │   500.0 │      5.93 │     90.1% │ 102.25 │     3.26 │       69.3%│
│    8 │ 191 │ 55563.4 │ 64175.0 │   500.0 │   500.0 │      6.56 │     90.8% │  53.36 │     3.27 │       69.4%│
│   16 │ 370 │ 55581.4 │ 64094.0 │   500.0 │   500.0 │      6.38 │     91.4% │  31.24 │     3.25 │       69.2%│
└──────┴─────┴─────────┴─────────┴─────────┴─────────┴───────────┴───────────┴────────┴──────────┴────────────┘
```

### Accuracy
- After optimization
```
2026-07-20 07:24:27 - evalscope - INFO: Overall report table: 
+---------------+-----------+----------+----------+-------+---------+---------+
| Model         | Dataset   | Metric   | Subset   |   Num |   Score | Cat.0   |
+===============+===========+==========+==========+=======+=========+=========+
| GLM-5.2-NVFP4 | aime26    | mean_acc | default  |    30 |  0.9333 | default |
+---------------+-----------+----------+----------+-------+---------+---------+ 
```

### CODE REVIEW

**Comment by borontion** [comment]
@Max191 can you also take a look of this optimization? thanks!
